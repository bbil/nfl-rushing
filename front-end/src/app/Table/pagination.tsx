import React from 'react';
import { connect } from 'react-redux';
import { getHasNextPage, getHasPreviousPage } from '../../redux/rushing/selectors';
import { nextPage, previousPage } from '../../redux/rushing/thunk';
import { State } from '../../redux/store';

interface PaginationStateProps {
    hasPreviousPage: boolean;
    hasNextPage: boolean;
}

interface PaginationDispatchProps {
    previousPage: () => void;
    nextPage: () => void;
}

export type PaginationProps = PaginationStateProps & PaginationDispatchProps;

export const PaginationComponent: React.FC<PaginationProps> = props => {
    if (!props.hasNextPage && !props.hasPreviousPage) {
        return null;
    }

    return (
        <div className="pagination">
            <button disabled={!props.hasPreviousPage} onClick={props.previousPage}>Previous</button>
            <button disabled={!props.hasNextPage} onClick={props.nextPage}>Next</button>
        </div>
    );
};

function mapStateToProps(state: State): PaginationStateProps {
    return {
        hasNextPage: getHasNextPage(state),
        hasPreviousPage: getHasPreviousPage(state)
    }
}

function mapDispatchToProps(dispatch: Function): PaginationDispatchProps {
    return {
        nextPage: () => dispatch(nextPage()),
        previousPage: () => dispatch(previousPage())
    }
}

export const Pagination = connect(mapStateToProps, mapDispatchToProps)(PaginationComponent);
