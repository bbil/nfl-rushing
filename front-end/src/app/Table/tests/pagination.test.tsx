import '@testing-library/react';

import { shallow } from 'enzyme';
import { PaginationComponent, PaginationProps } from '../pagination';

describe('Pagination component', () => {

    const PROPS: PaginationProps = {
        nextPage: jest.fn(),
        previousPage: jest.fn(),
        hasNextPage: false,
        hasPreviousPage: false
    };

    it('is null if no next page or previous page', () => {
        const wrapper = shallow(<PaginationComponent {...PROPS} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with only next page enabled', () => {
        const wrapper = shallow(<PaginationComponent {...PROPS} hasNextPage />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with only previous page enabled', () => {
        const wrapper = shallow(<PaginationComponent {...PROPS} hasPreviousPage />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with only both enabled', () => {
        const wrapper = shallow(<PaginationComponent {...PROPS} hasPreviousPage hasNextPage />);

        expect(wrapper).toMatchSnapshot();
    });
});