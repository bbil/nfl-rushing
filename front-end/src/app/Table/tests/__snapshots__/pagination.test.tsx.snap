// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component is null if no next page or previous page 1`] = `""`;

exports[`Pagination component renders with only both enabled 1`] = `
<div
  className="pagination"
>
  <button
    disabled={false}
    onClick={[MockFunction]}
  >
    Previous
  </button>
  <button
    disabled={false}
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;

exports[`Pagination component renders with only next page enabled 1`] = `
<div
  className="pagination"
>
  <button
    disabled={true}
    onClick={[MockFunction]}
  >
    Previous
  </button>
  <button
    disabled={false}
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;

exports[`Pagination component renders with only previous page enabled 1`] = `
<div
  className="pagination"
>
  <button
    disabled={false}
    onClick={[MockFunction]}
  >
    Previous
  </button>
  <button
    disabled={true}
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
